resolwe/flow/filters.py

```diff
@@ -163,6 +163,22 @@ def filter_for_user(self, queryset, name, value):
         user = self.request.user
         return queryset.filter_for_user(user, Permission.from_name(value))
 
+    def filter_for_group(self, queryset, name, value):
+        """Filter queryset by group."""
+        return queryset.filter(permission_group__permissions__group=value)
+
+    def filter_private(self, queryset, name, value):
+        """Return only elements that are not public (explicitly shared with me)."""
+
+        public_user = get_user_model().objects.get(username="public")
+        public_filter = Q(permission_group__permissions__user=public_user)
+
+        # if filter value is False, return only public elements
+        if value:
+            return queryset.filter(~public_filter)
```

Comment:
I am not sure but maybe it would be nicer to use exclude here instead of negating the Q object.